main/boofcv-ip/src/test/java/boofcv/alg/misc/impl/TestImplImageBandMath_MT.java

```diff
@@ -0,0 +1,301 @@
+package boofcv.alg.misc.impl;
+
+import boofcv.alg.misc.GImageMiscOps;
+import boofcv.concurrency.BoofConcurrency;
+import boofcv.core.image.FactoryGImageGray;
+import boofcv.core.image.GImageGray;
+import boofcv.core.image.GeneralizedImageOps;
+import boofcv.struct.image.ImageGray;
+import boofcv.struct.image.Planar;
+import boofcv.struct.image.*;
+import org.junit.jupiter.api.Test;
+import org.omg.SendingContext.RunTime;
+
+import java.util.Random;
+
+import static boofcv.alg.misc.impl.ImplImageBandMath.average;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+class TestImplImageBandMath_MT {
+	int width = 10;
+	int height = 15;
+	int numBands = 11;
+	int firstBand = 0;
+	int lastBand = numBands -1;
+	Class[] types = new Class[]{GrayU8.class, GrayU16.class, GrayS16.class, GrayS32.class, GrayS64.class, GrayF32.class, GrayF64.class};
+
+
+	Random rand = new Random(234);
+
+	@Test
+	void implement() {
+		for (Class type : types) {
+			testType(type);
+		}
+	}
+
+
+	void testType(Class type){
+		Planar input = new Planar(type, width, height,numBands);
+		ImageGray output = GeneralizedImageOps.createSingleBand(type, width, height);
+		ImageGray outputMT = GeneralizedImageOps.createSingleBand(type, width, height);
+		ImageGray avg = GeneralizedImageOps.createSingleBand(type, width, height);
+
+		if( output.getDataType().isSigned() ) {
+			GImageMiscOps.fillUniform(input, rand, -20,20);
+		} else {
+			GImageMiscOps.fillUniform(input, rand, 0,20);
+		}
+		GImageGray[] testImages = new GImageGray[lastBand - firstBand + 1];
+		for (int i = firstBand; i <= lastBand; i++) {
+			testImages[i] = FactoryGImageGray.wrap(input.getBand(i));
+		}
+		if (type == GrayU8.class) {
+			BoofConcurrency.USE_CONCURRENT = false;
```

Comment:
No need to toggle BoofConcurrency.USE_CONCURRENT.  Any call to ImplImageBandMath_MT will be concurrent. Which is why the simple compare function is sufficient.

Reply:
OK, removed.  I was not completely sure of the mechanism , but this makes sense.  Committed.